src/en/mangahot/src/eu/kanade/tachiyomi/extension/en/mangahot/MangaHot.kt

```diff
@@ -0,0 +1,310 @@
+package eu.kanade.tachiyomi.extension.en.mangahot
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.decodeFromStream
+import kotlinx.serialization.json.put
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import okio.Buffer
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+import java.net.URLEncoder
+import kotlin.math.ceil
+
+class MangaHot : HttpSource() {
+
+    override val name = "MangaHot"
+
+    override val baseUrl = "https://mangahot.to"
+
+    override val lang = "en"
+
+    override val supportsLatest = false
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(2)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder().apply {
+        add("Referer", "$baseUrl/")
+    }
+
+    private val apiHeaders = headersBuilder().apply {
+        add("Accept", "*/*")
+        set("Referer", "$baseUrl/list")
+        add("Sec-Fetch-Dest", "empty")
+        add("Sec-Fetch-Mode", "cors")
+        add("Sec-Fetch-Site", "same-origin")
+    }.build()
```

Comment:
save insteance of apiHeadersBuilder as well, and use it in place of apiHeaders.newBuilder() in the code.

Reply:
Do you need to do anything other than removing `.build()`? Because if I only do that, it's going to keep adding headers and setting certain headers that other requests don't normally have.

Reply:
https://github.com/keiyoushi/extensions-source/blob/374264cc79af519e1309f351febc1de19af8a96f/src/pt/yugenmangas/src/eu/kanade/tachiyomi/extension/pt/yugenmangas/YugenMangas.kt#L46

Something like this

Reply:
Turns out I forgot to make it a function instead of a variable. Woops